src/nameclaim.h

```diff
@@ -1,33 +1,45 @@
 #ifndef BITCOIN_NAMECLAIM_H
 #define BITCOIN_NAMECLAIM_H
 
+#include "amount.h"
 #include "script/script.h"
+#include "primitives/transaction.h"
 #include "uint256.h"
 
 #include <vector>
 
+// This is the minimum claim fee per character in the name of an OP_CLAIM_NAME command that must
+// be attached to transactions for it to be accepted into the memory pool.
+// Rationale: current implementation of the claim trie uses more memory for longer name claims
+// due to the fact that each chracater is assigned a trie node regardless of whether it contains
+// any claims or not. In the future, we can switch to a radix tree implementation where
+// empty nodes do not take up any memory and the minimum fee can be priced on a per claim
+// basis.
+#define MIN_FEE_PER_NAMECLAIM_CHAR 200000
```

Comment:
The PR looks good, and I haven't had as much time to consider the fee structure, but this strikes me as a bit high for a minimum at this point ... although it'll effectively act as a deterrent (perhaps the intended effect).